src/index.ml

```diff
@@ -159,6 +159,27 @@ struct
   let check_open t =
     match !t with Some instance -> instance | None -> raise Closed
 
+  let to_list t = List.of_seq (Tbl.to_seq t)
+
+  let io_to_list msg io =
+    let x = to_list io.mem in
+    let y = ref [] in
+    iter_io (fun (e : Entry.t) -> y := (e.key, e.value) :: !y) io.io;
+    if List.length x <> List.length !y then (
+      let pp_entry ppf (k, _) = pp_key ppf k in
+      let pp = Fmt.Dump.list pp_entry in
+      Fmt.epr "consistency error in %s:\nmem : %a\ndisk: %a\n%!" msg pp x pp !y;
+      assert false);
```

Comment:
After reading this again: probably not the best API here - it works only if the io instance has been flushed to disk before calling the function. I'll try to make it clearer.